### PR TITLE
fix: uses session name for routing

### LINKED ISSUE #58: Add dynamic DestinationRule Destination Subset config
Currently, hardcoded to "v1-test".

Possibly rely on session name as a suffix to the target v.

Affects `mutateDestinationRule` and `mutateVirtualService`

### PR DESCRIPTION
**Pull request by bartoszmajsak**
Fixes #58 

Bonus: godoc and var names cleanup

### CODE REVIEW

File: pkg/controller/session/model_convert.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 
 const (
 	// DefaultRouteHeaderName holds the name of the Header used to route traffic if no Route is provided
-	DefaultRouteHeaderName = "X-Workspace-Route"
+	DefaultRouteHeaderName = "x-workspace-route"
```
Comment (bartoszmajsak) [inline pkg/controller/session/model_convert.go]: According to istio docs header matchers should be lower case

> The header keys must be lowercase and use hyphen as the separator, e.g. `x-request-id`

Comment (bartoszmajsak) [reply pkg/controller/session/model_convert.go]: https://istio.io/docs/reference/config/networking/v1alpha3/virtual-service/#HTTPMatchRequest

File: pkg/istio/destinationrule.go [hunk 2]
```diff
@@ -64,7 +64,7 @@ func DestinationRuleRevertor(ctx model.SessionContext, ref *model.Ref) error { /
 		}
 
 		ctx.Log.Info("Found DestinationRule", "name", resource.Name)
-		mutatedDr, err := revertDestinationRule(*dr)
+		mutatedDr, err := revertDestinationRule(*dr, ctx)
```
Comment (aslakknutsen) [inline pkg/istio/destinationrule.go]: Why is revert taking 'ctx' while mutate just the 'name'?

Comment (bartoszmajsak) [reply pkg/istio/destinationrule.go]: Missed that, the code was changed multiple times where I was fiddling with the `ctx` to pass more metadata. That's clearly leftover. Fixed with 593618c

**Pull request merged**